core/src/main/scala/spinal/core/Misc.scala

```diff
@@ -133,6 +133,11 @@ object HardType{
     val width = elements.map(widthOf(_)).max
     HardType(Bits(width bits))
   }
+
+  def unionSeq(elements: Seq[Data]): HardType[Bits] = {
```

Comment:
There is a way to have both union and unionSeq being named union : 

```scala
  def union(head : Data, tail: Data*): HardType[Bits] =  ...
  def union(elements: Seq[Data]): HardType[Bits] = ...
```

Should be good ^^

Reply:
Too much rust made me forget this is possible in other languages :wink: